### PR TITLE
Update: no-multiple-define checks CallExpression only (fixes #42)

### LINKED ISSUE #42: no-multiple-define message could be more useful (and rule could be faster?)
The `no-multiple-define` rule works by iterating through all tokens in a program and reporting only once at the very end if the total number of `define` identifiers is greater than 1. This results in one message at the beginning of the file. I think the rule could be more useful if it reported all violating nodes (after the first one), so that line and column information is accurate.

In addition, the rule should not need to iterate over all tokens, only CallExpressions (after all, using the identifier `define` does not always imply a module definition, such as `typeof define === "function"`). By changing the rule to run its checks on CallExpression nodes, that should make the rule a bit faster. The rule can still keep track of define calls by closing over the count variable.

I'm happy to write a PR if you accept this issue.

### PR DESCRIPTION
**Pull request by platinumazure**
Also added a new test case where `define` appears multiple times but is only in a `CallExpression` once. This test case results in an error being reported with current master, but is valid with my code. If you consider that a breaking change rather than a bugfix, let me know and I can tag this as a breaking change.

### CODE REVIEW

File: tests/fixtures/index.js [hunk 1]
```diff
@@ -132,5 +132,8 @@ function loadFixture(path) {
     "./BAD_REQUIRE_INVALID_CALLBACK",
     "./BAD_REQUIRE_INVALID_ERRBACK",
     "./UNWRAPPED_FILE",
-    "./UNWRAPPED_FILE_NO_EXPRESSIONSTATEMENT"
+
+    // Edge cases that are actually valid
+    "./UNWRAPPED_FILE_NO_EXPRESSIONSTATEMENT",
```
Comment (platinumazure) [inline tests/fixtures/index.js:135]: Ah crap, the description above is wrong. UNWRAPPED_FILE_NO_EXPRESSIONSTATEMENT is invalid. I'll fix and push hopefully in the next day or so. Don't merge this yet.

**Comment by platinumazure** [comment]
Please don't merge this yet, I just saw a mistake. Will fix ASAP.

**Comment by cvisco** [comment]
No worries. I'll wait ;)

**Comment by platinumazure** [comment]
I'll have this fixed tonight. Sorry for the delay, work has been murder.

**Comment by platinumazure** [comment]
Should be good to merge @cvisco, unless you see any other problems. Thanks for your patience!

**Comment by cvisco** [comment]
Nice improvement!
